tendermint/src/lite/types.rs

```diff
@@ -96,7 +99,9 @@ pub trait Commit {
 /// is only necessary to avoid slashing in the multi chain context.
 pub trait Vote {
     fn validator_id(&self) -> Id;
-    fn sign_bytes(&self) -> &[u8];
+    fn sign_bytes<B>(&self, sign_bytes: &mut B)
+    where
+        B: BufMut;
```

Comment:
Maybe I did run into issues regarding the lifetime. I'll try to see if I can simply return ` &[u8]` as previously.

Reply:
Actually, the problem is that the slice we want return here is created from a temp value inside that  method (instead of sth borrowed from parameters of that method, in this case only `self`). e.g. this does not work:

```rust
    fn sign_bytes(&self) -> &[u8] {
        let mut sign_bytes = vec![];
        CanonicalVote::new(AminoVote::from(self), "TODO")
            .encode(&mut sign_bytes)
            .unwrap();
        sign_bytes.as_slice()
    }
```

On simple way to returning something would be to return a `Vec<u8>` instead of `&[u8]`. As it isn't reference this should work without any problems.
A much more complex alternative might be to add a field that caches the the sign bytes. Not sure the latter works or would be worth the effort (?).